lib/rubocop/cop/rspec/mock_not_stub.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Checks that message expectations are not combined with message stubs.
+      #
+      # @example
+      #
+      #   # bad
+      #   expect(foo).to receive(:bar).with(42).and_return("hello world")
+      #
+      #   # good (without spies)
+      #   allow(foo).to receive(:bar).with(42).and_return("hello world")
+      #   expect(foo).to receive(:bar).with(42)
```

Comment:
even better `expect(foo).to have_received(:bar).with(42)`

Reply:
The title says "without spies". Both would work with a partial double, but still `have_received` reminds of a spy-style.

Reply:
Somehow missed that, perhaps I should avoid such late evening reviews :)